R/methods-Collections.R

```diff
@@ -119,4 +156,35 @@ setMethod("pruneFeatures", signature("Collection"), function(object, predicate,
 
     validObject(object)
     return(object)
+})
+
+#' Is One to Many With Ancestor
+#' 
+#' Determines if the collection is one-to-many with its ancestor(s).
+#' Importantly, if there are no ancestors, this function returns FALSE.
+#' @param object Collection
+#' @param verbose boolean indicating if timed logging is desired
+#' @return boolean
+#' @rdname isOneToManyWithAncestor
+#' @export
+setGeneric("isOneToManyWithAncestor",
+    function(object, verbose = c(TRUE, FALSE)) standardGeneric("isOneToManyWithAncestor"),
+    signature = c("object")
+)
+
+#' @rdname isOneToManyWithAncestor
+#' @aliases isOneToManyWithAncestor,Collection-method
+setMethod("isOneToManyWithAncestor", signature("Collection"), function(object, verbose = c(TRUE, FALSE)) {
+    ancestorIdColumns <- object@ancestorIdColumns
+
+    if (length(ancestorIdColumns) == 0) return(FALSE)
+
+    # if count of unique ancestors are the same as ancestors, then its one-to-one
+    uniqueAncestors <- unique(object@data[, ..ancestorIdColumns])
```

Comment:
i dont understand what `unique` returns here. Let's say there are three ancestor id columns. The output is a data frame that keeps only entire rows that are unique? If we had two rows `c(a, b, c)` and `c(a, b, d)`, we'd keep both of these?

Reply:
correct. which should mean that if a sample had multiple assays done, and we ask only for ids from sample and above, that those unique combinations of ids will be fewer than the total number of rows..

Reply:
but doesn't that mean that if the cols are (householdid, sampleid) and we have two rows `c(a, b)` and `c(a, c)` that we'll see two unique entries, but actually the household isn't one to one with samples?